segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/MetadataTransaction.java

```diff
@@ -97,15 +97,23 @@
     @Setter
     private Callable<Void> externalCommitStep;
 
+    /**
+     * Array of keys to lock for this transaction.
+     */
+    @Getter
+    private final String[] keysToLock;
+
     /**
      * Constructor.
      *
      * @param store   Underlying metadata store.
      * @param version Version number of the transactions.
+     * @param keysToLock Array of keys to lock for this transaction.
      */
-    public MetadataTransaction(ChunkMetadataStore store, long version) {
+    public MetadataTransaction(ChunkMetadataStore store, long version, String... keysToLock) {
         this.store = Preconditions.checkNotNull(store, "store");
         this.version = version;
+        this.keysToLock = Preconditions.checkNotNull(keysToLock, "keys");
```

Comment:
If you invoke this as `new MetadataTransaction(store, version)` then `keysToUnlock` will be an empty array and never be null.

You should accept is as a String array (`String[] keysToLock`)

Reply:
added check.